File: frontend/types/conversation.ts
/**
 * Conversation-related TypeScript types for Phase 3 AI-powered Todo Chatbot.
 *
 * Constitutional Compliance: These types strictly follow the Phase 3 System Constitution.
 */

export interface Conversation {
  id: string;
  title: string | null;
  preview: string;
  created_at: string;
  updated_at: string;
}

// Backend message format (snake_case from API)
export interface BackendMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  tool_calls?: Record<string, any> | null;
  language?: string | null;
  created_at: string;
}

export interface ConversationDetail {
  id: string;
  user_id: string;
  title: string | null;
  messages: BackendMessage[];
  created_at: string;
  updated_at: string;
}

export interface ConversationListResponse {
  conversations: Conversation[];
  total: number;
}

File: backend/src/models/conversation.py
"""Conversation model for Phase 3 AI-powered Todo Chatbot.

This model represents a chat session between user and AI assistant.
"""

from sqlmodel import SQLModel, Field, Relationship
from datetime import datetime
from uuid import UUID, uuid4
from typing import Optional, List, TYPE_CHECKING

if TYPE_CHECKING:
    from .message import Message


class Conversation(SQLModel, table=True):
    """Conversation model representing a chat session.

    Constitutional Compliance: This model strictly follows the Phase 3 System Constitution.
    Database is the single source of truth for all conversation data.
    """

    __tablename__ = "conversations"

    id: UUID = Field(default_factory=uuid4, primary_key=True)
    user_id: UUID = Field(foreign_key="users.id", nullable=False, index=True)
    title: Optional[str] = Field(default=None, max_length=100)
    created_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
    updated_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)

    # Relationships
    messages: List["Message"] = Relationship(back_populates="conversation", cascade_delete=True)
